nfs/src/rpc_readdir.cpp

```diff
@@ -345,7 +346,7 @@ struct directory_entry *readdirectory_cache::lookup(
 
     const auto it = dir_entries.find(cookie);
 
-    struct directory_entry *dirent =
+    std::shared_ptr<struct directory_entry> dirent =
```

Comment:
this can be const shared_ptr& so that it refers to the stored directory_entry.
Anyways on return a fresh ref on the shared_ptr will be taken and returned to the caller.